pkg/didcomm/packager/package_test.go

```diff
@@ -0,0 +1,294 @@
+			return nil, fmt.Errorf("pack error")
+		}
+		mockPacker = &didcomm.MockAuthCrypt{EncryptValue: e}
+		mockedProviders.outboundPacker = mockPacker
+		packager, err = New(mockedProviders)
+		require.NoError(t, err)
+		packMsg, err = packager.PackMessage(&transport.Envelope{Message: []byte("msg1"),
+			FromVerKey: base58FromVerKey,
+			ToVerKeys:  []string{base58ToVerKey}})
+		require.Error(t, err)
+		require.Empty(t, packMsg)
+		require.EqualError(t, err, "pack: pack error")
+	})
+
+	t.Run("test Pack/Unpack success", func(t *testing.T) {
+		// create a mock KMS with storage as a map
+		w, err := kms.New(newMockKMSProvider(&mockstorage.MockStoreProvider{Store: &mockstorage.MockStore{
+			Store: map[string][]byte{}}}))
+		require.NoError(t, err)
+		mockedProviders := &mockProvider{
+			storage:        nil,
+			kms:            w,
+			outboundPacker: nil,
+			packers:        nil,
+		}
+
+		// create a real testPacker (no mocking here)
+		testPacker, err := jwe.New(mockedProviders, jwe.XC20P)
+		require.NoError(t, err)
+		mockedProviders.outboundPacker = testPacker
+
+		legacyPacker := legacy.New(mockedProviders)
+		mockedProviders.packers = []packer.Packer{testPacker, legacyPacker}
+
+		// now create a new packager with the above provider context
+		packager, err := New(mockedProviders)
+		require.NoError(t, err)
+
+		_, base58FromVerKey, err := w.CreateKeySet()
+		require.NoError(t, err)
+
+		_, base58ToVerKey, err := w.CreateKeySet()
+		require.NoError(t, err)
+
+		// pack an non empty envelope - should pass
+		packMsg, err := packager.PackMessage(&transport.Envelope{Message: []byte("msg1"),
+			FromVerKey: base58FromVerKey,
+			ToVerKeys:  []string{base58ToVerKey}})
+		require.NoError(t, err)
+
+		// unpack the packed message above - should pass and match the same payload (msg1)
+		unpackedMsg, err := packager.UnpackMessage(packMsg)
+		require.NoError(t, err)
+		require.Equal(t, unpackedMsg.Message, []byte("msg1"))
+
+		// pack with legacy, unpack using a packager that has JWE as default but supports legacy
+
+		mockedProviders.outboundPacker = legacyPacker
+
+		packager2, err := New(mockedProviders)
+		require.NoError(t, err)
+
+		packMsg, err = packager2.PackMessage(&transport.Envelope{Message: []byte("msg2"),
+			FromVerKey: base58FromVerKey,
+			ToVerKeys:  []string{base58ToVerKey}})
+		require.NoError(t, err)
+
+		// unpack the packed message above - should pass and match the same payload (msg2)
+		unpackedMsg, err = packager.UnpackMessage(packMsg)
+		require.NoError(t, err)
+		require.Equal(t, unpackedMsg.Message, []byte("msg2"))
+	})
+}
+
+func newMockKMSProvider(storagePvdr *mockstorage.MockStoreProvider) *mockProvider {
+	return &mockProvider{storagePvdr, nil, nil, nil}
+}
+
+// mockProvider mocks provider for KMS
+type mockProvider struct {
+	storage        *mockstorage.MockStoreProvider
+	kms            kms.KeyManager
+	packers        []packer.Packer
+	outboundPacker packer.Packer
```

Comment:
not sure `outbound`Packer has a meaning, it's just a `Packer` that should be able to do Pack/Unpack operations. 

why add the direction in the variable name?

Reply:
See reply below.

Reply:
I think we should combine these options into one to reduce confusion (and API surface).
 https://github.com/hyperledger/aries-framework-go/issues/764.